src/TrayIcon.cpp

```diff
@@ -119,7 +122,12 @@ TrayIcon::TrayIcon(const QString &filename, QWindow *parent)
     quitAction_ = new QAction(tr("Quit"), this);
 
     connect(viewAction_, &QAction::triggered, parent, &QWindow::show);
-    connect(quitAction_, &QAction::triggered, this, QApplication::quit);
+    connect(quitAction_, &QAction::triggered, this, [=] {
```

Comment:
You probably want to connect to aboutToQuit instead like here: https://github.com/Nheko-Reborn/nheko/blob/7f9af9016d2e5a5513457e191bc454885561a73a/src/main.cpp#L352